youtube_transcript_api/test/test_api.py

```diff
@@ -151,48 +153,71 @@ def test_get_transcript__exception_if_no_transcript_available(self):
+    def test_get_transcripts__stop_on_error(self, mock_get_transcript):
         with self.assertRaises(Exception):
             YouTubeTranscriptApi.get_transcripts(['video_id_1', 'video_id_2'])
 
-    def test_get_transcripts__continue_on_error(self):
+    @patch('youtube_transcript_api.YouTubeTranscriptApi.get_transcript', side_effect=Exception('Error'))
+    def test_get_transcripts__continue_on_error(self, mock_get_transcript):
         video_id_1 = 'video_id_1'
         video_id_2 = 'video_id_2'
-        YouTubeTranscriptApi.get_transcript = MagicMock(side_effect=Exception('Error'))
 
         YouTubeTranscriptApi.get_transcripts(['video_id_1', 'video_id_2'], continue_after_error=True)
 
-        YouTubeTranscriptApi.get_transcript.assert_any_call(video_id_1, ('en',), None)
-        YouTubeTranscriptApi.get_transcript.assert_any_call(video_id_2, ('en',), None)
-
-    def test_get_transcript__with_proxies(self):
+        mock_get_transcript.assert_any_call(video_id_1, ('en',), None, None)
+        mock_get_transcript.assert_any_call(video_id_2, ('en',), None, None)
+    
+    @patch('youtube_transcript_api.YouTubeTranscriptApi.get_transcript')
+    def test_get_transcripts__with_cookies(self, mock_get_transcript):
+        cookies = '/example_cookies.txt'
+        YouTubeTranscriptApi.get_transcripts(['GJLlxj_dtq8'], cookies=cookies)
+        mock_get_transcript.assert_any_call('GJLlxj_dtq8', ('en',), None, cookies)
+
+    @patch('youtube_transcript_api.YouTubeTranscriptApi.get_transcript')
+    def test_get_transcripts__with_proxies(self, mock_get_transcript):
         proxies = {'http': '', 'https:': ''}
-        transcript = YouTubeTranscriptApi.get_transcript(
-            'GJLlxj_dtq8', proxies=proxies
-        )
-
-        self.assertEqual(
-            transcript,
-            [
-                {'text': 'Hey, this is just a test', 'start': 0.0, 'duration': 1.54},
-                {'text': 'this is not the original transcript', 'start': 1.54, 'duration': 4.16},
-                {'text': 'just something shorter, I made up for testing', 'start': 5.7, 'duration': 3.239}
-            ]
-        )
-        YouTubeTranscriptApi.get_transcript = MagicMock()
         YouTubeTranscriptApi.get_transcripts(['GJLlxj_dtq8'], proxies=proxies)
-        YouTubeTranscriptApi.get_transcript.assert_any_call('GJLlxj_dtq8', ('en',), proxies)
+        mock_get_transcript.assert_any_call('GJLlxj_dtq8', ('en',), proxies, None)
+
+    def test_load_cookies(self):
+        dirname, filename = os.path.split(os.path.abspath(__file__))
+        cookies = dirname + '/example_cookies.txt'
+        session_cookies = YouTubeTranscriptApi.load_cookies(cookies)
+        self.assertEqual({'TEST_FIELD': 'TEST_VALUE'},  requests.utils.dict_from_cookiejar(session_cookies))
+
+    def test_load_cookies__bad_files(self):
+        bad_cookies = 'nonexistent_cookies.txt'
+        with self.assertRaises(Exception):
```

Comment:
Might as well test that it raises the correct Exception instead of just `Exception` to make sure that it isn't failing through something else.